### ISSUE TITLE
Inject Internet permission at build time

### ISSUE THREAD

**Issue by mit-mit** [flutter/flutter#8421]
When we create a new Flutter app, we add this to `AndroidManifest.xml`:

```
    <!-- The INTERNET permission is required for development. Specifically, flutter needs it to communicate with the running application
         to allow setting breakpoints, to provide hot reload, etc.
    -->
    <uses-permission android:name="android.permission.INTERNET"/>
```

Developers are expected to remove this when producing release builds if their app does not need this permission. However, as there is only a single manifest file used for both debug and release that means you need to edit the file whenever you are switching between debug and release.

It would be much better if we could inject this permission into the build at build time (for debug builds only!).

**Comment by eseidelGoogle** [comment]
See also https://github.com/flutter/flutter/issues/2138.

**Comment by zoechi** [comment]
Looks like this was fixed by #22139, #26450
